src/_posts/2019-02-25-dynamodb-explained-part-1.md

```diff
@@ -0,0 +1,62 @@
+
+Before getting started let's see where DynamoDB lives inside the database categories.
+
+![alt text]({{site.baseurl}}/assets/custom/img/blog/2019-02-25-dynamodb-explained-part-1/01-dynamodb-db-tree.jpg "DynamoDB inside the DB family")
+
+# What is DynamoDB?
+
+DynamoDB is a NoSQL, highly-available, low latency, scalable database that supports both key-value and document store data models. It follows a serverless approach, meaning that it is offered as a service, so the infrastructure is fully managed by AWS and works out-of-the-box with little configuration. This frees the user from managing instances, applying security patches and other infrastructure-related tasks. In addition, pricing can be very reasonable as it follows a pay-per-use model instead of charging for pre-provisioned computing resources.
+
+As there is no instance the database concept, existent on most relational databases and also in some NoSQL like MongoDB, disappears and all is done on tables. Moreover, tables are schemaless meaning that the objects stored are not required to have the same attributes, providing more flexibility to the user.
+
+DynamoDB shines when providing high performance for large workloadst as it handles read and write throughput independently. Because of this, it is aimed at analytical operations but it can still work as a cheap, quick, low configuration solution for simple CRUD applications.
+
+For ease of use, AWS provides an SDK for several languages including as Java, .NET, Node.js, PHP, Python, Ruby, Go, C++ and JavaScript for browsers and IoT devices. In addition to all that, requests can be made using JSON over HTTP.
+
+# What is it not?
+
+Although it is capable of handling documents, don't think of DynamoDB as a like-for-like replacement for MongoDB. It is built around the key-value data store model, so it bears more similarity to Redis or Memcached. 
+
+A key difference is that by default not all attributes of a document can be efficiently queried, only those that belong to a primary key or an index. This is because it's high performance comes from searching for a specific element by its key, rather than filtering on other attributes. Moreover, DynamoDB does not support querying embedded data structures, nor does it provide full-text search or other features like MapReduce or date functions.
+
+# The DynamoDB Primary Key Approach
+ 
+Like many databases, DynamoDB requires you  define a primary key that uniquely identifies an element in a table. DynamoDB uses primary keys slightly differently from some other databases as the key has a direct impact on how data is physically stored, which in turn is critical for it's performance, as will be explained in the second part of this series. Similar to relational databases, the primary key can be simple or composite, the latter being compound of two attributes.
```

Comment:
I think there is an `a` missing here 
`the latter being compound of two attributes`
vs
`the latter being a compound of two attributes`